### ISSUE TITLE
visible button in a sticky element in a modal is detected as not visible

### ISSUE THREAD

**Issue by FraKraBa** [cypress-io/cypress#14938]
### Current behavior
I have a button inside a sticky element in a modal. Behind the modal is a scrollable element which is an ancestor of the modal. Cypress thinks that the button is not visible. I get the following error message:

> This element `<button#button>` is not visible because its content is being clipped by one of its parent elements, which has a CSS property of overflow: `hidden`, `scroll` or `auto`

Here is the screenshot of the failing test:
![page -- works (failed)](https://user-images.githubusercontent.com/54527640/107015265-d883af80-679c-11eb-9e0d-25221ee1a39d.png)

### Desired behavior
I can see the button. Cypress should recognize that it is visible.

### Test code to reproduce
I have a simple example which reproduces this error. You can find it here:
https://github.com/FraKraBa/cypress-test-tiny

### Versions
Cypress 6.4.0

**Comment by FraKraBa** [comment]
I debugged a little bit and found the following suspicious function:
`
const elOrAncestorIsFixed = function ($el) {
    const $stickyOrFixedEl = $elements.getFirstFixedOrStickyPositionParent($el);

    if ($stickyOrFixedEl) {
       return $stickyOrFixedEl.css('position') === 'fixed'; // we are sticky => returns false
    }
};
`
In the case presented the button is inside a sticky element. Thus, `elOrAncestorIsFixed` returns false. Afterwards `elIsOutOfBoundsOfAncestorsOverflow` is called and tells that the button is not inside its ancestor Details which has `overflow:scroll`. However, between the button and Details is the model which hat the property `position:fixed`.

**Comment by jennifer-shehane** [comment]
@FraKraBa Thanks for the reproducible example and digging in to the cause. 

### Reproducible Example

`index.html`

```html
<html>
<body>
  <div id="parent" style="display:flex; height: 90vh;">
    <div id="Preselect" style="flex:1; background-color:aquamarine; height: 100%;">
      Preselect
    </div>
    <div id="Details" style="flex:1; background-color:blue; height: 100%; overflow:scroll;">
      Details
      <div id="modal" style="background-color: chartreuse; height: 80vh; width: 90%; position: fixed; display: block; left:5%; top:5%;">
        <div id="sticky" style="background-color: darkgray; position:sticky; bottom: 10%; width:9%; height:10%;">
          <button id="button" style="background-color: darkred; color:white;">
            click me
          </button>
          <div id="message"></div>
        </div>
      </div>
    </div>
  </div>
</body>
<script>
  const btn = document.getElementById('button')
  const msg = document.getElementById('message')

  btn.onclick = (e) => {
    msg.innerHTML = 'Success!'
  }
</script>

</html>
```

`spec.js`

```js
it('works', () => {
  cy.visit('index.html');
  cy.get('#button').should('be.visible');
})
```

![](http://g.recordit.co/lBeNgp53Sl.gif)

**Comment by bahmutov** [comment]
Released in v7.0.0

**Comment by TekSiDoT** [comment]
Sweet, thank you @FraKraBa and Cypress Team
